nservicebus/messaging/evolving-contracts.md

```diff
@@ -29,24 +26,92 @@ It's recommended to use a dedicated assembly for message definitions. By keeping
 
 It's also possible to share messages as C# source files without packaging them into an assembly. One advantage of this approach is that messages don't need to be compiled against specific NServiceBus versions, so assembly redirects are not necessary. This can be accomplished by using [unobstrusive mode messages](/nservicebus/messaging/unobtrusive-mode.md) as well.
 
-
 ## Common challenges
 
+### Handling breaking changes in contracts
 
-### Default values
+There are a number of approaches to handling breaking changes in contracts. One of the easiest methods is creating an entirely new contract type by copying the original and then adding a version number to the original name. Once the endpoint is upgraded to the new contracts assembly, create a new message handler for the new message. The message handler for the original message can still process the original message, both for in-flight messages and also for messages generated by endpoints that have not been upgraded yet to the new message contract.
 
-When a message is extended with additional properties, it's necessary to carefully examine what will be the default values for those properties, especially if endpoints running in other versions don't recognize them. In particular, it's important to consider how clients might interpret the default value and provide appropriate guidelines for them.
+For complex systems, especially when there are significant changes in a message type, such as adding or removing properties, or changing the property type, a step-by-step approach to evolving message contracts is recommended.
 
+#### Adding data to a contract
 
-### Handling breaking changes in contracts
+Depending on the complexity of the change, the number of senders and receivers for the contract as well as how the change might propagate through the system, a different approach may be desirable. It's important to consider where the added data is required for a message to be successfully processed.
+
+Considering a scenario in which the data that is added to the contract is not required to succesfully process a message:
+
+* Update the contract and release a new version.
+  * Instead of relying on .NET to set the default value for int Age = 1, it's better to use nullable types and represent missing values as null. This allows for message handlers to reliably verify if the data is available using the ´HasValue´-property.
+* Update senders to use the new contract version.
+* Update receivers to handle the new contract version. 
+  * This can be done gradually, endpoint by endpoint. 
```

Comment:
```suggestion
  * This can be done gradually, endpoint by endpoint. Endpoints that are not upgraded and reference the previous contracts assembly, will be unaware of the newly added data as it will be ignored during deserialization.
```

Reply:
@dvdstelt This was in my notes but I forgot to put it in the PR, do you think it makes sense to add this to further clarify?

Reply:
You can just merge this, but if you think it needs to be clarified further, then yeah.

Reply:
Sounds good!